findsecbugs-plugin/src/main/java/com/h3xstream/findsecbugs/xml/SchemaFactoryDetector.java

```diff
@@ -0,0 +1,164 @@
+import edu.umd.cs.findbugs.ba.AnalysisContext;
+import edu.umd.cs.findbugs.ba.CFG;
+import edu.umd.cs.findbugs.ba.CFGBuilderException;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import edu.umd.cs.findbugs.ba.Location;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+import org.apache.bcel.Const;
+import org.apache.bcel.generic.ConstantPoolGen;
+import org.apache.bcel.generic.ICONST;
+import org.apache.bcel.generic.INVOKEVIRTUAL;
+import org.apache.bcel.generic.Instruction;
+import org.apache.bcel.generic.InstructionHandle;
+import org.apache.bcel.generic.InvokeInstruction;
+import org.apache.bcel.generic.LDC;
+
+import javax.xml.XMLConstants;
+
+/**
+ * Detector for XML External Entity and External Schema processing in javax.xml.validation.SchemaFactory
+ */
+public class SchemaFactoryDetector extends OpcodeStackDetector {
+
+    private static final String XXE_SCHEMA_FACTORY_TYPE = "XXE_SCHEMA_FACTORY";
+
+    private static final String SCHEMA_FACTORY_CLASS_NAME = "javax/xml/validation/SchemaFactory";
+
+    private static final String SET_FEATURE_METHOD = "setFeature";
+
+    private static final String SET_PROPERTY_METHOD = "setProperty";
+
+    private static final String NEW_SCHEMA_METHOD_NAME = "newSchema";
+
+    private static final Number BOOLEAN_TRUE_VALUE = 1;
+
+    private static final String EXTERNAL_REFERENCES_DISABLED = "";
+
+    private final BugReporter bugReporter;
+
+    public SchemaFactoryDetector(final BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+    }
+
+    @Override
+    public void sawOpcode(final int opcode) {
+        if (isNotNewSchemaMethod(opcode)) {
+            return;
+        }
+
+        final ClassContext classContext = getClassContext();
+        final CFG cfg;
+        try {
+            cfg = classContext.getCFG(getMethod());
+        } catch (final CFGBuilderException e) {
+            AnalysisContext.logError("Cannot get CFG", e);
+            return;
+        }
+
+        final ConstantPoolGen cpg = classContext.getConstantPoolGen();
+
+        boolean secureProcessingEnabled = false;
+        boolean accessExternalDtdDisabled = false;
+        boolean accessExternalSchemaDisabled = false;
+
+        for (final Location location : cfg.locations()) {
+            if (isSecureProcessingEnabled(location, cpg)) {
+                secureProcessingEnabled = true;
+            } else if (isAccessPropertyDisabled(location, cpg, XMLConstants.ACCESS_EXTERNAL_DTD)) {
+                accessExternalDtdDisabled = true;
+            } else if (isAccessPropertyDisabled(location, cpg, XMLConstants.ACCESS_EXTERNAL_SCHEMA)) {
+                accessExternalSchemaDisabled = true;
+            }
+        }
+
+        // Enabling Secure Processing or disabling both Access External DTD and Access External Schema are solutions
+        if (secureProcessingEnabled || (accessExternalDtdDisabled && accessExternalSchemaDisabled)) {
+            return;
+        }
```

Comment:
The return instruction in the middle is probably not super popular but here it makes the condition very readable.